packages/node-core/src/indexer/dictionary.service.ts

```diff
@@ -140,12 +142,28 @@ export class DictionaryService implements OnApplicationShutdown {
 
   constructor(
     readonly dictionaryEndpoint: string,
+    readonly chainId: string,
     protected readonly nodeConfig: NodeConfig,
     protected readonly metadataKeys = ['lastProcessedHeight', 'genesisHash'] // Cosmos uses chain instead of genesisHash
-  ) {
+  ) {}
+
+  async init(): Promise<void> {
+    let link: ApolloLink;
+
+    if (this.nodeConfig.authDictionary) {
+      assert(this.dictionaryEndpoint, 'dictionary endpoint must be included in manifest to use --authDictionary');
+      link = await authHttpLink({
```

Comment:
Do we need to handle errors this could throw?

Reply:
updated